src/test/HostActivationTests/NativeHosting/Nethost.cs

```diff
@@ -0,0 +1,122 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.Cli.Build;
+using Microsoft.DotNet.Cli.Build.Framework;
+using System;
+using System.IO;
+using Xunit;
+
+namespace Microsoft.DotNet.CoreSetup.Test.HostActivation.NativeHosting
+{
+    public class Nethost : IClassFixture<Nethost.SharedTestState>
+    {
+        private const string GetHostFxrPath = "nethost_get_hostfxr_path";
+
+        private readonly SharedTestState sharedState;
+        private readonly DotNetCli dotNetCli;
+
+        public Nethost(SharedTestState sharedTestState)
+        {
+            sharedState = sharedTestState;
+            dotNetCli = new DotNetCli(Path.Combine(TestArtifact.TestArtifactsPath, "sharedFrameworkPublish"));
+        }
+
+        [Fact]
+        public void GetHostFxrPath_NoAssemblyPath_NoFxr()
+        {
+            Command.Create(sharedState.NativeHostPath, GetHostFxrPath)
+                .CaptureStdErr()
+                .CaptureStdOut()
+                .Execute()
+                .Should().Fail();
+        }
+
+        [Fact]
+        public void GetHostFxrPath_NoAssemblyPath_FxrSubdirectory()
+        {
+            string copiedHostPath = Path.Combine(Path.GetDirectoryName(sharedState.NativeHostPath), "host");
+            SharedFramework.CopyDirectory(Path.Combine(dotNetCli.BinPath, "host"), copiedHostPath);
```

Comment:
This will probably go away, but if we keep it in some form: The test pollutes the directory used by other tests - it puts the hostfxr next to the native host... and keeps it there for the other tests.

You need some kind of try/finally to delete it - or create a copy in a new directory just for this test.

Reply:
This went away, but I ended up making the shared test state just set everything up separately.